src/popup/bookmarkModule.js

```diff
@@ -302,18 +302,29 @@ document.addEventListener('DOMContentLoaded', () => {
     removeBookmarkImages();
   });
 
-  elements.deleteAllBookmarksButton.addEventListener('click', () => {
+  elements.deleteBookmarksButton.addEventListener('click', () => {
     chrome.storage.sync.get({ bookmarks: [] }, (items) => {
       const bookmarksArray = items.bookmarks;
-      if (bookmarksArray.length === 0) {
-        showNotification('No Bookmarks Available', 'negative', 'snackbar-bookmarks');
+      const bookmarkDOMArray = Object.values(bookmarkDOM);
+      bookmarkDOMArray.forEach((checkbox) => {
+        if (checkbox.checked) {
+          bookmarksArray[bookmarksArray.indexOf(checkbox.id)] = null;
+        }
+      });
+      const updatedBookmarksArray = [];
+      bookmarksArray.forEach((bookmark) => {
+        if (bookmark != null) {
+          updatedBookmarksArray.push(bookmark);
+        }
+      });
+      if (bookmarksArray.length === updatedBookmarksArray.length) {
+        showNotification('No bookmark selected', 'negative', 'snackbar-bookmarks');
       } else {
-        bookmarksArray.splice(0, bookmarksArray.length); // empty array
-        chrome.storage.sync.set({ bookmarks: bookmarksArray }, () => {
+        chrome.storage.sync.set({ bookmarks: updatedBookmarksArray }, () => {
           // restoring initial layout of bookmarks section
           removeBookmarkImages();
           msnry.layout();
-          restoreInitialContent('bookmarks');
+          loadImages();
```

Comment:
@harsh551 What you have done is remove the loading spinner, therefore it looks like the images are not loaded again. But, since you are using the `loadImages()` function, they are indeed being reloaded after every deletion.  As I have said above, we have to implement something like [this](https://codepen.io/desandro/pen/IBJeq) (i.e don't reload the images after deletion but let the masonry grid reorganize them). This is what happens when you remove the images by clicking the bookmark icon (gif below). [This function](https://github.com/creativecommons/ccsearch-browser-extension/blob/master/src/popup/bookmarkModule.js#L45) implements that.

![ezgif com-crop (2)](https://user-images.githubusercontent.com/34679965/76159894-00acfd80-614b-11ea-98ba-1bf017774170.gif)